pages/api/flags/[projectId].ts

```diff
@@ -29,7 +45,13 @@ const handle = async (req, res) => {
   });
 
   res.setHeader("Cache-Control", "s-maxage=60, stale-while-revalidate");
-  res.status(200).json(flags);
+  res
+    .status(200)
+    .json(
+      flags.filter((flag) =>
+        flag.heats.every((heat) => isHeatApplicable(heat, req.query.config))
```

Comment:
Not sure, would probably be easier to really unify to something with as little parsing and encoding on both sides😀 I did not get what is the suggestion😕 I will try to experiment with this and see how it looks like when you send something.

Reply:
We can just send the config as the body of the request btw. Wdyt?

Reply:
Body cannot be sent in GET request and use a POST is an anti pattern as the call only retrieve information and didn't create any entity

Reply:
Technically we handle that internally so we might as well adopt the option with dynamic query params construction right? To the react lib we just pass an object with desired config 

```
<Provider config={{env:"dev",user:"123"}} />
```

Than it would boil down to your example in the request to flags api with query string 

`?env=dev&user=123`

Which could easily be handled on the level of the backend of the application.

Do we want to handle dynamic and static configs separately though? Something like environment could be static, user role for example could be dynamic. Will we need to handle this in a special way?

Thoughts @alfredosalzillo ?

Reply:
To have something more solid to speak about:

```
function isHeatApplicable(customHeat: Heat, config: Map<string, string>) {
  const value = config[customHeat.property];
  if (value) {
    switch (customHeat.strategy) {
      case "IN":
        return customHeat.values.includes(value);
    }
    switch (customHeat.strategy) {
      case "NOT_IN":
        return !customHeat.values.includes(value);
    }
  }
  return false;
}

const handle = async (req, res) => {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  const { projectId, ...config } = req.query;

  const flags = await prisma.featureFlag.findMany({
    where: {
      projectId: String(req.query.projectId),
    },
    include: {
      heats: true,
    },
  });

  res.setHeader("Cache-Control", "s-maxage=60, stale-while-revalidate");
  res
    .status(200)
    .json(
      flags.filter((flag) =>
        flag.heats.every((heat) => isHeatApplicable(heat, config))
      )
    );
};
```

Reply:
just noticed I have 2 parallel switch statements lol 😆  obviously needs some refactoring but you get the overall idea of how we could set the contract between the API and react-lib

Reply:
I think this way it's good :)